system/modules/core/widgets/FileSelector.php

```diff
@@ -457,7 +457,7 @@ protected function convertValuesToPaths()
 		}
 
 		// Do not convert values if path is not in the upload path
-		if ($this->path !== \Config::get('uploadPath'))
+		if (strncmp($this->path, \Config::get('uploadPath'), strlen(\Config::get('uploadPath'))) !== 0)
```

Comment:
To make it 100% accurate, it should be:

```php
if ($this->path != \Config::get('uploadPath') && strpos($this->path, \Config::get('uploadPath').'/') !== 0)
```

Otherwise the condition will evaluate to `true` if `$this->path` is e.g. `files-and-docs`.